lib/user_agent/browsers/samsung.rb

```diff
@@ -0,0 +1,35 @@
+class UserAgent
+  module Browsers
+    class Samsung < Base
+
+      SAMSUNG_BROWSER = 'SamsungBrowser'
+
+      def self.extend?(agent)
+        agent.detect { |useragent| useragent.product == SAMSUNG_BROWSER}
+      end
+
+      def version
+        self.detect { |useragent| useragent.product == SAMSUNG_BROWSER}.version
+      end
+
+      ##
+      # @return [Boolean]
+      #     SamsungBrowser is a mobile device
+      def mobile?
+        true
+      end
+
+      def browser
+        SAMSUNG_BROWSER
```

Comment:
```suggestion
        'Samsung Internet Browser'
```

This seems to be the name that Samsung prefers (according to here: https://play.google.com/store/apps/details?id=com.sec.android.app.sbrowser)